tableauserverclient/server/endpoint/endpoint.py

```diff
@@ -65,6 +76,54 @@ def set_user_agent(parameters):
         # return explicitly for testing only
         return parameters
 
+    def _blocking_request(self, method, url, parameters={}) -> Optional["Response"]:
+        self.async_response = None
+        response = None
+        logger.debug("[{}] Begin blocking request to {}".format(datetime.timestamp(), url))
+        try:
+            response = method(url, **parameters)
+            self.async_response = response
+            logger.debug("[{}] Call finished".format(datetime.timestamp()))
+        except Exception as e:
+            logger.debug("Error making request to server: {}".format(e))
+            self.async_response = e
+        finally:
+            if response and not self.async_response:
+                logger.debug("Request response not saved")
+                return None
+        logger.debug("[{}] Request complete".format(datetime.timestamp()))
+        return self.async_response
+
+    def _user_friendly_blocking_request(self, method, url, parameters={}, request_timeout=0) -> Optional["Response"]:
+        try:
+            request_thread = Thread(target=self._blocking_request, args=(method, url, parameters))
+            request_thread.async_response = -1  # type:ignore # this is an invented attribute for thread comms
+            request_thread.start()
+        except Exception as e:
+            logger.debug("Error starting server request on separate thread: {}".format(e))
+            return None
+        seconds = 0
+        minutes = 0
+        sleep(1)
+        if self.async_response != -1:
+            # a quick return for any immediate responses
+            return self.async_response
+        while self.async_response == -1 and (request_timeout == 0 or seconds < request_timeout):
+            self.wait_for_response(minutes, seconds, url)
+            seconds = seconds + DELAY_SLEEP_SECONDS
+            if seconds >= 60:
+                seconds = 0
+                minutes = minutes + 1
+        return self.async_response
+
+    def wait_for_response(self, minutes, seconds, url):
+        logger.debug("{} Waiting....".format(datetime.timestamp()))
+        if minutes % 5 == 0 and seconds >= 60:
+            logger.info("[{}] Waiting ({} minutes so far) for request to {}".format(datetime.timestamp(), minutes, url))
+        elif minutes % 1 == 0 and seconds >= 60:
```

Comment:
Was this supposed to be `minutes == 1`?

Reply:
no, originally I was just logging every minute and then I added the longer message every 5 minutes and borked the checks